xdem/spatial_tools.py

```diff
@@ -396,3 +398,102 @@ def hillshade(dem: Union[np.ndarray, np.ma.masked_array], resolution: Union[floa
     # Return the hillshade, scaled to uint8 ranges.
     # The output is scaled by "(x + 0.6) / 1.84" to make it more similar to GDAL.
     return np.clip(255 * (shaded + 0.6) / 1.84, 0, 255).astype("float32")
+
+
+@numba.njit
+def _get_closest_rectangle(size: int) -> tuple[int, int]:
+    """
+    Given a 1D array size, return a rectangular shape that is closest to a cube which the size fits in.
+
+    If 'size' does not have an integer root, a rectangle is returned that is slightly larger than 'size'.
+    
+    :examples:
+        >>> _get_closest_rectangle(4)  # size will be 4
+        (2, 2)
+        >>> _get_closest_rectangle(9)  # size will be 9
+        (3, 3)
+        >>> _get_closest_rectangle(3)  # size will be 4; needs padding afterward.
+        (2, 2)
+        >>> _get_closest_rectangle(55) # size will be 56; needs padding afterward.
+        (8, 7)
+        >>> _get_closest_rectangle(24)  # size will be 25; needs padding afterward
+        (5, 5)
+        >>> _get_closest_rectangle(85620)  # size will be 85849; needs padding afterward
+        (293, 293)
+        >>> _get_closest_rectangle(52011)  # size will be 52212; needs padding afterward
+        (229, 228)
+    """
+    close_cube = int(np.sqrt(size))
+
+    if close_cube ** 2 == size:
+        return (close_cube, close_cube)
+
+    height = close_cube
+    width = close_cube
+    # Iteratively increase height, then width, until the size is larger or equal to the input size
+    for i in range(int(1e8)):
+        if i % 2 == 0:
+            height += 1
+        else:
+            width += 1
+
+        if (height * width) >= size:
+            return (height, width)
+
+    else:
+        raise ValueError("Close rectangle could not be found after 1e8 iterations.")
+
+
+
+
+def subdivide_array(shape: tuple[int, ...], count: int) -> np.ndarray:
```

Comment:
FYI: 
Commonly used block splitter/stitcher set of functions for numpy arrays to use with multiprocessing: https://github.com/iamdonovan/pyddem/blob/master/pyddem/fit_tools.py#L1366
Same for overlapping blocks (patchify/unpatchify):
https://github.com/iamdonovan/pyddem/blob/master/pyddem/fit_tools.py#L1389
But we should not reinvent the wheel on this, and probably switch to functions developed efficiently in Dask at some later point! :)

Reply:
They look very simple, which would be nice! I must however admit that I have no idea what's going on in them since there are no comments. Might you explain what the inputs/outputs look like?